vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -35,6 +36,17 @@ class Action(VizroBaseModel):
         regex="^[a-zA-Z0-9_]+[.][a-zA-Z_]+$",
     )
 
+    # TODO: Abstract away once pattern becomes clear regarding function/import checks
+    @validator("function")
+    def check_export_imports(cls, function):
+        if (
+            function._CapturedCallable__function.__name__ == "export_data"
+            and function._arguments.get("file_format") == "xlsx"
+        ):
+            if importlib.util.find_spec("openpyxl") is None and importlib.util.find_spec("xlsxwriter") is None:
```

Comment:
Do you have any reference on why to check for modules this way? I've never seen it before and the Python docs are not particularly enlightening.

What I was expecting/probably more conventional was `import` wrapped in `try/except`, but I expect the `importlib` version is fine if it works.

Reply:
Good point - so this was how I was implementing it first, but then `ruff` complained and suggested this alternative way, which I agree is the cleaner way.

https://docs.astral.sh/ruff/rules/unused-import/

Reply:
Ah nice 👍 Thanks for the information.